typescript/infra/scripts/squads/cancel-proposal.ts

```diff
@@ -0,0 +1,212 @@
+import { confirm } from '@inquirer/prompts';
+import chalk from 'chalk';
+import yargs from 'yargs';
+
+import { SvmMultiProtocolSignerAdapter } from '@hyperlane-xyz/sdk';
+import { ProtocolType, rootLogger } from '@hyperlane-xyz/utils';
+
+import { chainsToSkip } from '../../src/config/chain.js';
+import {
+  buildSquadsProposalCancellation,
+  buildSquadsProposalRejection,
+  getSquadProposal,
+  withTransactionIndex,
+} from '../../src/utils/squads.js';
+import { getTurnkeySealevelDeployerSigner } from '../../src/utils/turnkey.js';
+import { chainIsProtocol } from '../../src/utils/utils.js';
+import { withChain } from '../agent-utils.js';
+import { getEnvironmentConfig } from '../core-utils.js';
+
+const environment = 'mainnet3';
+
+// CLI argument parsing
+async function main() {
+  const { chain, transactionIndex } = await withTransactionIndex(
+    withChain(yargs(process.argv.slice(2))),
+  ).demandOption('chain').argv;
+
+  // Validate chain is Sealevel
+  if (!chainIsProtocol(chain, ProtocolType.Sealevel)) {
+    throw new Error(
+      `Chain ${chain} is not a Sealevel chain. This script only works with Solana/SVM chains.`,
+    );
+  }
+
+  if (chainsToSkip.includes(chain)) {
+    throw new Error(
+      `Chain ${chain} is in the skip list and cannot be used for Squads operations.`,
+    );
+  }
+
+  // Get multi-protocol provider
+  const envConfig = getEnvironmentConfig(environment);
+  const mpp = await envConfig.getMultiProtocolProvider();
+
+  rootLogger.info(
+    chalk.cyan(
+      `\n=== Processing Squads Proposal ${transactionIndex} on ${chain} ===`,
+    ),
+  );
+
+  // Fetch the proposal to verify it exists
+  const proposalData = await getSquadProposal(chain, mpp, transactionIndex);
+  if (!proposalData) {
+    throw new Error(
+      `Proposal ${transactionIndex} not found on ${chain}. Please check the transaction index.`,
+    );
+  }
+
+  const { proposal, multisig } = proposalData;
+  const status = proposal.status.__kind;
+  rootLogger.info(chalk.gray(`Found proposal with status: ${status}`));
+
+  // Check if proposal is stale
+  const staleTransactionIndex = Number(multisig.staleTransactionIndex);
+  if (transactionIndex < staleTransactionIndex) {
+    rootLogger.warn(
+      chalk.yellow(
+        `⚠️  Proposal ${transactionIndex} is stale (current stale index: ${staleTransactionIndex})`,
+      ),
+    );
+    rootLogger.info(
+      chalk.gray(
+        'Stale proposals can still be rejected/cancelled but will not affect voting.',
+      ),
+    );
+  }
+
+  // Check if proposal is already executed, cancelled, or rejected
+  if (status === 'Executed') {
+    throw new Error(
+      `Proposal ${transactionIndex} has already been executed and cannot be modified.`,
+    );
+  }
+
+  if (status === 'Cancelled') {
```

Comment:
do we have the status strings somewhere else too? if we do maybe as a drive by you could have them be an enum to avoid magic strings

Reply:
ok, devs did something
<img width="396" height="421" alt="image" src="https://github.com/user-attachments/assets/4e4b90e2-c321-4981-9839-3add280cff18" />